src/Nethermind/Nethermind.Synchronization/FastSync/TreeSync.cs

```diff
@@ -741,7 +741,11 @@ private void CleanupMemory()
 
         private void StoreProgressInDb()
         {
-            byte[] serializedData = _data.Serialize();
+            byte[]? serializedData;
+            lock (_data)
```

Comment:
Are there any other locks on `_data` where this will effect (e.g. does it stop modifications of `_data`)?

Could it just move under the WriteLock?

Reply:
yes, my bad, this will not do anything.
i though this will block access to other write operations also, my bad.
would create a ReaderWriterLockSlim for this variable and use it whenever it is modified.